addons/im_livechat/static/src/embed/core/livechat_service.js

```diff
@@ -198,8 +230,11 @@ export class LivechatService {
         return Boolean(this.cookie.current[this.SESSION_COOKIE]);
     }
 
-    get shouldDeleteSession() {
-        return this.sessionCookie && this.sessionCookie.visitor_uid !== session.user_id;
+    /**
+     * @returns {string|undefined}
+     */
+    get guestToken() {
+        return this.sessionCookie?.guest_token;
```

Comment:
To do in follow-up PR: clean the overuse of getters in livechat code.
This is not great having so many getters, and they have nested call to other getters.
You should define clear source of truth, and limit use on this object for all things related to it.

If you rely on `sessionCookie`, always rely on this single `cookieCookie`. Show explicitly in all part of code that `sessionCookie?.guest_token`, don't hide it in the getter.